pkg/storagev2/migration/migration_test.go

```diff
@@ -51,25 +51,101 @@ func TestGetSetVersion(t *testing.T) {
 
 func TestValidateVersions(t *testing.T) {
 	t.Parallel()
-
-	// TODO
+	tests := []struct {
+		name    string
+		input   StepsMap
+		wantErr bool
+	}{
+		{
+			name: "missing version 3",
+			input: StepsMap{
+				1: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "aaa", SomeInt: 1})
+				},
+				2: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "bbb", SomeInt: 2})
+				},
+				4: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "ccc", SomeInt: 3})
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "not missing",
+			input: StepsMap{
+				1: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "aaa", SomeInt: 1})
+				},
+				2: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "bbb", SomeInt: 2})
+				},
+				3: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "ccc", SomeInt: 3})
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "desc order versions",
+			input: StepsMap{
+				3: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "aaa", SomeInt: 1})
+				},
+				2: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "bbb", SomeInt: 2})
+				},
+				1: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "ccc", SomeInt: 3})
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "desc order version missing",
+			input: StepsMap{
+				4: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "aaa", SomeInt: 1})
+				},
+				2: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "bbb", SomeInt: 2})
+				},
+				1: func(s storage.Store) error {
+					return s.Put(&obj1{Id: "ccc", SomeInt: 3})
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if err := ValidateVersions(tt.input); (err != nil) != tt.wantErr {
+				t.Errorf("ValidateVersions() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
 }
 
 func TestMigrate(t *testing.T) {
 	t.Parallel()
+	objT1 := &obj1{Id: "aaa", SomeInt: 1}
+	objT2 := &obj1{Id: "bbb", SomeInt: 2}
+	objT3 := &obj1{Id: "ccc", SomeInt: 3}
```

Comment:
👍

Perhaps have same for others tests?

Reply:
Yes. Done